### ISSUE TITLE
Converting .jpeg to .exr 

### ISSUE THREAD

**Issue by hblasins** [benjamin-heasly/mexximp#3]
The jpeg to exr conversion does not work.

1. The mexximpExrTools has a parameter 'dockerImage'. This parameter is normally called 'exrtoolsImage'
2. There are some flaws in logic handling paths to different directories. This becomes apparent when the model is placed in a directory in the working folder.

**Comment by benjamin-heasly** [comment]
Could you attach a specific example to help me reproduce the problem you're seeing?  That would make it easier for me to fix the bug.

**Comment by hblasins** [comment]
I've created a separate branch to work in this issue (exrFix). 
1. I've fixed the parameter naming conversion in mexximpExrTools.m.
2. I modified the code in mexximpRecodeImage so that texture files stored in multi-level sub directories are correctly read and converted, but:
   - I'm not sure this fix is generic enough
   - The converted exr files are saved in the working directory rather than the sub-directory where they came from. This is to allow RTB4 to automatically copy all the necessary files to the resources folder in the directory with rendering outputs.

**Comment by benjamin-heasly** [comment]
Thanks for the branch!  I will look soon and also run some tests to make sure existing code works with the change.

Could you make a PR for this branch?  I think that will be a nice way to handle comments and revisions, and merging.

**Comment by hblasins** [comment]
Hi Ben,

I don’t want to create a PR just yet. Still working on some details.

Also, I’ve come across a mesh that uses textures in .TGA format. I’ll update the mexximpExrTools function to support this file type as well. I’ll probably create a new docker container with the relevant tools.

Cheers,
Henryk

 On Nov 16, 2016, at 7:50 AM, Ben Heasly <notifications@github.com<mailto:notifications@github.com>> wrote:

Thanks for the branch! I will look soon and also run some tests to make sure existing code works with the change.

Could you make a PR for this branch? I think that will be a nice way to handle comments and revisions, and merging.

**Comment by benjamin-heasly** [comment]
Hi @hblasins -- I made a PR to help me look at the changes.  But you can still make changes to the branch and the PR should update itself.

**Comment by benjamin-heasly** [comment]
For the parameter name in `mexximpExrTools.m`, I think you're right.  Thanks for making the change.  I think we should likewise change all occurrences in the comments and variables.  Maybe with find and change `dockerImage` -> `exrtoolsImage`?

For .tga support, I think this is great.  I assume this new "convert" operation will only work with imagemagic, and not plain old exrtools?  If so, I think we should update the comments to say that "convert" is special and has a different dependency.

For locating image files in `mexximpRecodeImage`, I think we want to be able to locate files in multi-level subdirectories, like you say.  But I think the current code at [line 85](https://github.com/RenderToolbox/mexximp/blob/e4fb73898aa39c4176adf4f0847135e2fed7d93d/utilities/import/mexximpRecodeImage.m#L85) 
```
    originalPath = fullfile(workingFolder, sceneFolder, imageFile);
```
will produce a bad path when `workingFolder` and `sceneFolder` both have their default value of `pwd()`.

I think `sceneFolder` is the input and `workingFolder` is the output.  In that interpretation, I think the way to locate files in multi-level subdirectories would be to do a search within the `sceneFolder`.  How about that?

**Comment by hblasins** [comment]
@benjamin-heasly 

1. I'll have a look and the naming and change `dockerImage` to `exrtoolsImage` for consistency.

2. The new docker image I've been using also has exrTools installed, so it is backwards compatible. I was thinking about using `imagemagic` for all conversion operations, but I have yet to test it on `.exr` image conversion. In addition the new docker image is substantially larger; 1GB vs 400MB or so. But maybe for simplicity and consistency we should move to `imagemagic`? What do you think?

3. Yep, the file location code was not fully tested...

**Comment by benjamin-heasly** [comment]
Hi @hblasins --

1. Sweet.

2. I think if `imagemagic` does the job with exr files, then we should go ahead and move to it.  I agree the simplicity would be good, and worth the image size.  Do you want to do some EXR tests?

3. Cool.  Fair enough.  Do you want to have a short Skype chat this week to figure out what to do with `sceneFolder` and `imageFolder` and who should make the next changes?

**Comment by hblasins** [comment]
Hi Ben,

how about chatting on Skype on Wednesday at 12 (noon) PST?

Best,
Henryk


Wiadomość napisana przez Ben Heasly <notifications@github.com<mailto:notifications@github.com>> w dniu 05.12.2016, o godz. 09:00:


Hi @hblasins<https://github.com/hblasins> --

  1.  Sweet.

  2.  I think if imagemagic does the job with exr files, then we should go ahead and move to it. I agree the simplicity would be good, and worth the image size. Do you want to do some EXR tests?

  3.  Cool. Fair enough. Do you want to have a short Skype chat this week to figure out what to do with sceneFolder and imageFolder and who should make the next changes?

**Comment by benjamin-heasly** [comment]
@hblasins OK, see you tomorrow at noon aka 3pm.

**Comment by hblasins** [comment]
Hi Ben,

just wanted to let you know that I may be a few minutes late…

Also, so I don’t forget; I’m working with a very large mesh right now (its a model of a city) and it takes RTB4 hours to generate a PBRT file. To give you the idea of the size, about 75k geometry files are being generated during the export. Is this time constraint something I’ll have to live with, or could things be accelerated?

Henryk



Wiadomość napisana przez Ben Heasly <notifications@github.com<mailto:notifications@github.com>> w dniu 06.12.2016, o godz. 11:29:


@hblasins<https://github.com/hblasins> OK, see you tomorrow at noon aka 3pm.

**Comment by benjamin-heasly** [comment]
I don't know if we can speed that up or not.  I haven't tried working with large scenes like that before.  I suggest you make a separate issue about that.

**Comment by benjamin-heasly** [comment]
Merged PR #9 to close this.
